v.alkis.buildings.import.py

```diff
@@ -0,0 +1,250 @@
+# % options: Brandenburg,Berlin,Baden-Württemberg,Bayern,Bremen,Hessen,Hamburg,Mecklenburg-Vorpommern,Niedersachsen,Nordrhein-Westfalen,Rheinland-Pfalz,Schleswig-Holstein,Saarland,Sachsen,Sachsen-Anhalt
+# %end
+
+# %flag
+# % key: r
+# % description: Restrict ALKIS building data import to current region
+# %end
+
+# %rules
+# % required: federal_state, file
+# %end
+
+# %rules
+# % excludes: file, federal_state
+# %end
+
+# %rules
+# % excludes: aoi_map, -r
+# %end
+
+
+import requests
+import zipfile
+import py7zr
+import os
+import sys
+import atexit
+from io import BytesIO
+import grass.script as grass
+
+sys.path.insert(
+    1,
+    os.path.join(
+        os.path.dirname(sys.path[0]), "etc", "v.alkis.buildings.import"
+    ),
+)
+from download_urls import URLS, filenames
+
+
+rm_vec = []
+path_to_tempdir = None
+temp_region = None
+output_alkis_temp = None
+pid = None
+
+
+def cleanup():
+    # remove unzipped federal state data in tempdir-folder
+    if path_to_tempdir:
+        if os.path.exists(path_to_tempdir):
+            grass.try_rmdir(path_to_tempdir)
+    # if aoi_map is given:
+    if temp_region:
+        # set region back and delete saved region:
+        grass.run_command("g.region", region=temp_region)
+        grass.run_command(
+            "g.remove", type="region", name=temp_region, flags="f", quiet=True
+        )
+        # remove temp_output (if aoi_map given)
+        grass.run_command(
+            "g.remove",
+            type="vector",
+            name=output_alkis_temp,
+            flags="f",
+            quiet=True,
+        )
+
+
+def main():
+    global rm_vec, path_to_tempdir, temp_region, output_alkis_temp, pid
+    pid = os.getpid()
+    """ parser options:
+    """
+    aoi_map = options["aoi_map"]
+    file_federal_state = options["file"]
+    load_region = flags["r"]
+    """ get federal state
+    """
+    if file_federal_state:
+        with open(file_federal_state) as f:
+            federal_states = f.read()
+    else:
+        federal_states = options["federal_state"]
+    """ get URL for corresponding federal state
+    """
+    URL = None
+    fs = None
+    for federal_state in federal_states.split(","):
+        if federal_state in URLS:
+            if federal_state in ["Nordrhein-Westfalen", "Berlin"]:
+                URL = URLS[federal_state]
+                fs = federal_state
+            else:
+                grass.warning(
+                    _(f"Support for {federal_state} is not yet implemented.")
+                )
+        else:
+            if options["file"]:
+                grass.fatal(
+                    _(
+                        "Non valid name of federal state,"
+                        " in 'file'-option given"
+                    )
+                )
+            elif options["federal_state"]:
+                grass.fatal(
+                    _(
+                        "Non valid name of federal state,"
+                        " in 'federal_states'-option given"
```

Comment:
```suggestion
                        " in 'file' parameter given"
                    )
                )
            elif options["federal_state"]:
                grass.fatal(
                    _(
                        "Non valid name of federal state,"
                        " in 'federal_states' parameter given"
```

(minor wording suggestion, feel free to ignore)